enhancements/microshift/audit-log-configuration-options.md

```diff
@@ -0,0 +1,234 @@
+
+**Included Dependencies**
+
+- **openshift/library-go**: already imported by MicroShift.
+- **openshift/api**: will be added to MicroShift's direct dependency list.
+
+### Risks and Mitigations
+
+- Exceeding disk capacity: Microshift targets small form-factor devices with limit on-board storage. If the product of `maxFileSize` and `maxFiles` equals a size larger than the available storage, the apiserver risks destabilizing the system by maxing out the consumed storage.  This can be mitigated via documentation which recommends the user understand their storage limitations when setting these values. Logging a warning during startup when the apiserver values exceed the available storage at the logging path. It is impossible to predict which component of MicroShift or the host system will fail when the disk is out of space, thus logging a calculated overage can provide an important clue.
+- Lost Log Data: The apiserver culls log files given a certain size or age. If users do not take care to back up logs at a rate greater than the rate at which the kube-apiserver culls the files, they risk losing log data. This can be mitigated with examples of log-forwarding provided in documentation, example manifests, or both. Alternatively, it may be necessary to consider deploying the openshift-cluster-logging-operator to provide supportable log forwarding features.
+- Exposing the apiserver's audit-log-path, which allows users to set a custom log location, would hinder sos report gathering. Instead, users should replace `/var/log/kube-apiserver` with a symlink the desired path.  Sos will follow the symlink and collect logs by default.
+
+### Drawbacks
+
+- N/A
+
+## Open Questions
+
+- Custom log paths: Can sos collect reports from dynamic paths? If not, can we symlink the log directory to as a means of faking a static path while writing logs to a custom location?  If so, can sos follow these symlinks?
+
+    - SOS can traverse symlinks to collect log files from a target directory.  The link itself and the target path are included in the tarball, meaning support can easily locate audit logs via the default location (`/var/log/kube-apiserver`).
+
+- Should Microshift support a `minFreeStorage` field that would be used to determine whether an acceptable amount of space exists at the log file path? This is not a value recognized by the apiserver, but it could be used by Microshift to check if the system is in an acceptable state to boot into.  If the free space is lower than the value, MicroShift would not start and log a fatal error to document the reason. If unset, MicroShift should deploy regardless of the available storage.
+
+    - Providing additional fields and internal handling logic is outside the scope of this EP. If this is a useful feature, it should be documented in a bespoke EP.
+
+## Test Plan
+
+* Unit tests to validate new config APIs
+* Integration tests to verify configured audit logging policies are applied properly
```

Comment:
We should add a test to ensure we're solving the issue described in https://redhat-internal.slack.com/archives/C03DP9PABNC/p1708586503890529 where a service on the cluster is overriding the iptable rule for the router.

Reply:
Ugh, it was.